src/main/java/http/request/RequestParser.java

```diff
@@ -1,15 +1,39 @@
 package http.request;
 
+import http.helper.Messages;
+import http.helper.Utils;
 
+import java.io.*;
+import java.nio.Buffer;
 
-import java.nio.charset.StandardCharsets;
-import java.util.*;
 
 public class RequestParser {
 
 
+    public static void requestReader(BufferedReader reader) throws IOException {
 
+        Utils.print(Messages.httpReceived(reader.readLine()));
+        Utils.print("output read");
 
 
+    }
+
+
+    public static String[] parseRequest(String requestLine) {
+        String[] splitData = requestLine.split(" ");
+        return splitData;
+    }
+
+    public static Request buildRequest(String method, String path, String protocol) {
+        Request request = new Request();
+        request.setMethod(method);
+        request.setPath(path);
+        request.setHeaders("protocol", protocol);
+        return request;
+
+    }
```

Comment:
Have you considered connecting the two functions and calling just one in `Router`? in essence, parsing means understanding something and transforming it into something else (in your case the requestLine to the request object).

Reply:
I didn't want it to be doing too much in one function.